File: mobile/src/__tests__/shared/constants/medColors.test.ts
import {
  medColorSoftFill,
  resolveMedColorHex,
} from '@/shared/constants/medColors';

describe('medColorSoftFill', () => {
  it('설정색 hex를 흐린 rgba로', () => {
    const hex = resolveMedColorHex('coral');
    expect(hex).toBe('#C46B5A');
    expect(medColorSoftFill('coral', 0.2)).toBe('rgba(196, 107, 90, 0.2)');
  });

  it('기본 alpha 0.2', () => {
    expect(medColorSoftFill('teal')).toBe('rgba(77, 134, 121, 0.2)');
  });
});

File: android-app/core/src/test/kotlin/com/jinlabs/yakok/core/constants/MedColorsTest.kt
package com.jinlabs.yakok.core.constants

import kotlin.test.assertEquals
import org.junit.jupiter.api.Test

class MedColorsTest {
    @Test
    fun softFill_coral() {
        assertEquals("#C46B5A", MedColors.resolveHex("coral"))
        assertEquals("rgba(196, 107, 90, 0.2)", MedColors.softFill("coral", 0.2f))
    }

    @Test
    fun softFill_defaultAlpha() {
        assertEquals("rgba(77, 134, 121, 0.2)", MedColors.softFill("teal"))
    }

    @Test
    fun unknownFallsBackToTeal() {
        assertEquals("#4D8679", MedColors.resolveHex("nope"))
    }
}
